Crash report 3.7: NPE in AppLog.LogEntry.toHtml

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.AppLogViewerActivity}: java.lang.NullPointerException
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2282)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2340)
       at android.app.ActivityThread.access$800(ActivityThread.java:157)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1247)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:157)
       at android.app.ActivityThread.main(ActivityThread.java:5293)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1265)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1081)
       at dalvik.system.NativeStart.main(NativeStart.java)
Caused by: java.lang.NullPointerException
       at android.text.TextUtils.htmlEncode(TextUtils.java:1361)
       at org.wordpress.android.util.AppLog$LogEntry.toHtml(AppLog.java:121)
       at org.wordpress.android.util.AppLog$LogEntry.access$400(AppLog.java:115)
       at org.wordpress.android.util.AppLog.toHtmlList(AppLog.java:186)
       at org.wordpress.android.ui.AppLogViewerActivity$LogAdapter.(AppLogViewerActivity.java:55)
       at org.wordpress.android.ui.AppLogViewerActivity$LogAdapter.(AppLogViewerActivity.java:50)
       at org.wordpress.android.ui.AppLogViewerActivity.onCreate(AppLogViewerActivity.java:47)
       at android.app.Activity.performCreate(Activity.java:5394)
       at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1105)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2246)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2340)
       at android.app.ActivityThread.access$800(ActivityThread.java:157)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1247)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:157)
       at android.app.ActivityThread.main(ActivityThread.java:5293)
       at java.lang.reflect.Method.invokeNative(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:515)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1265)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1081)
       at dalvik.system.NativeStart.main(NativeStart.java)
```

--- a/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/AppLog.java
+++ b/libs/utils/WordPressUtils/src/main/java/org/wordpress/android/util/AppLog.java
@@ -113,22 +113,31 @@ public class AppLog {
     }
 
     private static class LogEntry {
-        LogLevel logLevel;
-        String logText;
-        T logTag;
+        LogLevel mLogLevel;
+        String mLogText;
+        T mLogTag;
+
+        public LogEntry(LogLevel logLevel, String logText, T logTag) {
+            mLogLevel = logLevel;
+            mLogText = logText;
+            if (mLogText == null) {
+                mLogText = "null";
+            }
+            mLogTag = logTag;
+        }
 
         private String toHtml() {
-            StringBuilder sb = new StringBuilder()
-                    .append("<font color=\"")
-                    .append(logLevel.toHtmlColor())
-                    .append("\">")
-                    .append("[")
-                    .append(logTag.name())
-                    .append("] ")
-                    .append(logLevel.name())
-                    .append(": ")
-                    .append(TextUtils.htmlEncode(logText).replace("\n", "<br />"))
-                    .append("</font>");
+            StringBuilder sb = new StringBuilder();
+            sb.append("<font color=\"");
+            sb.append(mLogLevel.toHtmlColor());
+            sb.append("\">");
+            sb.append("[");
+            sb.append(mLogTag.name());
+            sb.append("] ");
+            sb.append(mLogLevel.name());
+            sb.append(": ");
+            sb.append(TextUtils.htmlEncode(mLogText).replace("\n", "<br />"));
+            sb.append("</font>");
             return sb.toString();
         }
     }
@@ -155,12 +164,10 @@ public class AppLog {
 
     private static void addEntry(T tag, LogLevel level, String text) {
         // skip if recording is disabled (default)
-        if (!mEnableRecording)
+        if (!mEnableRecording) {
             return;
-        LogEntry entry = new LogEntry();
-        entry.logLevel = level;
-        entry.logText = text;
-        entry.logTag = tag;
+        }
+        LogEntry entry = new LogEntry(level, text, tag);
         mLogEntries.addEntry(entry);
     }
 
@@ -169,7 +176,6 @@ public class AppLog {
         throwable.printStackTrace(new PrintWriter(errors));
         return errors.toString();
     }
-
 
     /*
      * returns entire log as html for display (see AppLogViewerActivity)
@@ -188,7 +194,6 @@ public class AppLog {
         return items;
     }
 
-
     /*
      * returns entire log as plain text
      */
@@ -203,7 +208,7 @@ public class AppLog {
         int lineNum = 1;
         while (it.hasNext()) {
               sb.append(String.format("%02d - ", lineNum))
-              .append(it.next().logText)
+              .append(it.next().mLogText)
               .append("\n");
             lineNum++;
         }

